core/src/main/java/com/microsoft/applicationinsights/TelemetryConfiguration.java

```diff
@@ -77,6 +80,33 @@ public static TelemetryConfiguration getActive() {
     }
 
     /**
+     * Set {@link TelemetryConfiguration} instance to be 'Active'.
+     * Telemetry configuration must not be set twice explicitly or through calling {@link TelemetryConfiguration#getActive()}.
+     *
+     * @param telemetryConfiguration preconfigured instance of Telemetry Configuration
+     */
+    public static void setActive(TelemetryConfiguration telemetryConfiguration) {
+        synchronized (s_lock) {
+            if (initialized) {
+                throw new IllegalStateException("Telemetry configuration is already initialized. " +
```

Comment:
Public APIs do not throw in Application Insights SDK

Reply:
I agree with Sergey's point here. Throwing an exception in public API will crash the customers application and prevent it from starting. It is not a desired behavior.

Reply:
I would agree with you about runtime exceptions, but this one is startup exception, where, from my point of view, is better to get exception when something goes wrong rather than found that telemetry is not working after application is started. But if this is a strong sdk convention, I can just log error into logs.